docs/structured-streaming-programming-guide.md

```diff
@@ -2251,13 +2251,13 @@ When the streaming query is started, Spark calls the function or the object’s
 
 - The close() method (if it exists) is called if an open() method exists and returns successfully (irrespective of the return value), except if the JVM or Python process crashes in the middle.
 
-- **Note:** The partitionId and epochId in the open() method can be used to deduplicate generated data 
-  when failures cause reprocessing of some input data. This depends on the execution mode of the query. 
-  If the streaming query is being executed in the micro-batch mode, then every partition represented 
-  by a unique tuple (partition_id, epoch_id) is guaranteed to have the same data. 
-  Hence, (partition_id, epoch_id) can be used to deduplicate and/or transactionally commit 
-  data and achieve exactly-once guarantees. However, if the streaming query is being executed 
-  in the continuous mode, then this guarantee does not hold and therefore should not be used for deduplication.
+- **Note:** Spark doesn't guarantee same output for (partitionId, epochId) on failure, so deduplication
```

Comment:
no big deal but I usually avoid abbreviation in the doc. `doesn't` -> `does not`

Reply:
Makes sense. Majority of usages in the doc is "does not" instead of "doesn't". Thanks for guiding :)